test/waypoint_test.jsx

```diff
@@ -871,14 +871,22 @@ describe('<Waypoint>', function() {
       expect(this.subject).not.toThrow();
     });
 
-    xit('errors when multiple children are provided', () => {
+    it('errors when multiple children are provided', () => {
+      // INFO: React 16 throws user errors again by it's own in dev mode, see:
+      // https://github.com/facebook/react/issues/10384#issuecomment-334142138
+      //
+      // This code ignores uncaught errors and prevents test from failing in React 16
+      window.onerror = () => undefined;
+
       this.props.children = [
         <div key={1} />,
         <div key={2} />,
         <div key={3} />,
       ];
 
       expect(this.subject).toThrowError(notValidErrorMessage);
+
+      window.onerror = undefined;
```

Comment:
If window.onerror had been previously set to something else, this will break it. It would be a good idea to store the original value before assigning it, and then set it back to that here.

Reply:
Yep, forgot about that! Fixed :)